### PR TITLE
new style for group in user show page

### PR DESCRIPTION
**Pull request by Andrii-Kozak**
image
image
image

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -33,6 +33,10 @@ def update
     end
   end
 
+  def groups
+    @groups = @user.groups.page params[:page]
```
Comment (pereslop) [inline app/controllers/users_controller.rb:37]: add brackets

Comment (Andrii-Kozak) [reply app/controllers/users_controller.rb]: ok

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -33,6 +33,10 @@ def update
     end
   end
 
+  def groups
+    @groups = @user.groups.page(params[:page])
```
Comment (pereslop) [inline app/controllers/users_controller.rb:37]: check console if you don't have n+1 queries, when you load this page, and add screenshot to this PR

File: app/views/users/_group.html.slim [hunk 3]
```diff
@@ -0,0 +1,12 @@
+.col-md-6
```
Comment (pereslop) [inline app/views/users/_group.html.slim:1]: why you  add this partial, I think you can reuse from groups folder

File: app/views/users/groups.html.slim [hunk 4]
```diff
@@ -0,0 +1,8 @@
+- provide(:title, 'My groups')
```
Comment (pereslop) [inline app/views/users/groups.html.slim:1]: do you have this wording on user/1/groups page?

File: app/views/users/groups.html.slim [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+.jumbotron
+  h1
+    | #{current_user.first_name} Groups
```
Comment (pereslop) [inline app/views/users/groups.html.slim:4]: move this to 'provide' method at line 1

File: app/views/users/show.html.slim [hunk 6]
```diff
@@ -20,10 +20,20 @@
...
-        .card.mt-3[style="height: 220px;"]
-          | Friends
+        .card.mt-3.p-2
+          = link_to groups_user_path, class:"link-user-show" do
+            h2.link-user-show.mb-2.mt-2
+              | My Groups
+          end
+          .row
+            - @user.groups.limit(4).each do |group|
+              = render 'users/group', group: group
```
Comment (pereslop) [inline app/views/users/show.html.slim:30]: you need to reuse partial from groups folder

File: spec/controllers/users_controller_spec.rb [hunk 7]
```diff
@@ -90,4 +90,11 @@
       expect(flash[:danger]).to be_present
     end
   end
+
+  context 'when user controller routes' do
+    it 'render users#groups' do
```
Comment (pereslop) [inline spec/controllers/users_controller_spec.rb:95]: good

**Review by pereslop — COMMENTED** [review]
check N+1, and reuse groups/_group partial

File: app/views/users/groups.html.slim (hunk 5 above)
Comment (pereslop) [reply app/views/users/groups.html.slim:4]: should it be current user? what if I opened groups for other user, it will still show logined user name

Comment (Andrii-Kozak) [reply app/views/users/groups.html.slim:4]: fixed

File: app/views/users/_group.html.slim (hunk 3 above)
Comment (Andrii-Kozak) [reply app/views/users/_group.html.slim:1]: in this partial other styles

**Pull request merged**